src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/InterceptedProjectProperties/BuildPropertyPage/TrimModeValueProvider.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Properties;
+
+[ExportInterceptingPropertyValueProvider("TrimMode", ExportInterceptingPropertyValueProviderFile.ProjectFile)]
+internal class TrimModeValueProvider : InterceptingPropertyValueProviderBase
+{
+    public override async Task<string?> OnSetPropertyValueAsync(string propertyName, string unevaluatedPropertyValue, IProjectProperties defaultProperties, IReadOnlyDictionary<string, string>? dimensionalConditions = null)
+    {
+        // When setting the TrimMode, we also need to set the PublishTrimmed property to true.
+        if (string.Equals(unevaluatedPropertyValue, "full", StringComparison.OrdinalIgnoreCase))
+        {
+            await defaultProperties.SetPropertyValueAsync("PublishTrimmed", "true");
+        }
+        else
+        {
+            // If the user sets the TrimMode to anything other than full, set the PublishTrimmed property to false.
+            await defaultProperties.SetPropertyValueAsync("PublishTrimmed", "false");
```

Comment:
Do we surface "PublishTrimmed" as a separate checkbox in the UI?

Reply:
IMO, I think we should show "Publish Trimmed" as a checkbox and not show "TrimMode" in the UI. If you select "Publish Trimmed" or "Publish AOT", you get "Full" trimming and we discourage changing it to anything else. If neither trimmed nor AOT are selected, you don't get any trimming.

cc @agocke @vitek-karas @DamianEdwards

Reply:
Is there a use case for something other than "Full"? That is, would we ever want to guide users to select anything other than "Full"?

Reply:
@sayedihashimi please weigh in here.

Currently as per the documentation TrimMode has 2 options Full and Partial. Checkbox would be good if all we want to do is to set it to Full. For partial we would need to provide additional info in : TrimmableAssembly (which we dont currently have a way to specify it in the UI)

Reply:
We don't want to promote `partial` as an option. It is a legacy trimming option used in Maui and Blazor WASM. See https://github.com/dotnet/sdk/issues/30059#issuecomment-1563804107 and https://github.com/dotnet/aspnetcore/issues/48416#issuecomment-1562955642

@agocke - can we get the documentation updated to mention this?

Reply:
> IMO, I think we should show "Publish Trimmed" as a checkbox and not show "TrimMode" in the UI. If you select "Publish Trimmed" or "Publish AOT", you get "Full" trimming and we discourage changing it to anything else. If neither trimmed nor AOT are selected, you don't get any trimming.
> 
> cc @agocke @vitek-karas @DamianEdwards

@DamianEdwards what do you think of this proposal? It seems to align with our objectives?

Reply:
The challenge is with Blazor WebAssembly and ASP.NET Core projects prior to .NET 8, which both default to "Partial" trimming. The design of this UI was such that the properties were separate but that they didn't offer the chance to select "Partial" (inline with our guidance) but would display it if that were the actual value.

Reply:
I agree with Eric - I would not surface the `TrimMode` to the users via UX. The SDK will default to the right `TrimMode` when `PublishTrimmed=true` - that is for Blazor it will default to `partial` and for console it will default to `full` (in .net 8 that is). Changing this should be considered very advanced and honestly asking people to do so via editing the `csproj` file directly is the least of their worries.

I don't know how much value there is in letting people know what `TrimMode` is used. Personally, I would even shy away from showing any trim-related UX for Blazor apps, mainly because the behavior is so different from the console apps. Also, can the UI actually tell what trim mode will be used - it would have to run the build to determine that...

The main difference between the two modes for typical developers will be the trim warnings - they will be produced for the console app and won't be produced for the Blazor app. But that behavior is actually not affected by `TrimMode`.

Reply:
That's fair. Since this was designed we changed the ASP.NET Core default to be "full" so exposing the level is less important now.

Reply:
Thanks @DamianEdwards, @vitek-karas,  @sayedihashimi , @eerhardt . So if I understand correctly:
- Do not explose TrimMode
- Add checkbox for : Publish Trimmed. Default it to unchecked (?)
- Set PublishTrimmed to true when checked.
- Do not set TrimMode under the hood when the PublishTrimmed property is set since SDK sets to the right TrimMode
Anything else?

Reply:
The summary looks correct to me. @DamianEdwards anything else to add here, we are finalizing this PR soon.

Reply:
That looks good!

Reply:
Looks good to me too. Thanks!

Reply:
Great..thanks all

Reply:
Thanks for the feedback here; I've gone ahead and pushed some changes, UI looks like this now:
![image](https://github.com/dotnet/project-system/assets/8518253/cb76e7fb-016d-4ba5-9539-1c3d890374fb)

Reply:
I took what our docs had as descriptions, but let me know if you'd like something else.